### PR TITLE
Bug/jenkins 44222 missing external link to branch

### PR DESCRIPTION
**Pull request by NicuPascu**
# Description
Adds missing external link to branch

See [JENKINS-44222](https://issues.jenkins-ci.org/browse/JENKINS-44222).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by michaelneale** [comment]
nice one @NicuPascu - I will take a look some time. The filures may be intermittent, but one of them is worth looking into (looking at the ATH failures in the log).

**Comment by i386** [comment]
change looks good to me 🐝

**Comment by michaelneale** [comment]
I am seeing a :bug: 

There has been an error while trying to get the data. TypeError: Cannot read property 'url' of null
    at RunDetailsHeader.render (jenkins-js-extension.js:47252)
    at ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext (blueocean.js:84968)
    at ReactCompositeComponentWrapper._renderValidatedComponent (blueocean.js:84991)
    at ReactCompositeComponentWrapper.performInitialMount (blueocean.js:84533)
    at ReactCompositeComponentWrapper.mountComponent (blueocean.js:84429)
    at Object.mountComponent (blueocean.js:91442)
    at ReactDOMComponent.mountChildren (blueocean.js:90320)
    at ReactDOMComponent._createInitialChildren (blueocean.js:85896)
    at ReactDOMComponent.mountComponent (blueocean.js:85721)
    at Object.mountComponent (blueocean.js:91442)

When I try to open some runs from a favourite ( I think it is the header)

```
]jenkins-js-extension.js:47252 Uncaught TypeError: Cannot read property 'url' of null
    at RunDetailsHeader.render (jenkins-js-extension.js:47252)
    at ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext (blueocean.js:84968)
    at ReactCompositeComponentWrapper._renderValidatedComponent (blueocean.js:84991)
    at ReactCompositeComponentWrapper.performInitialMount (blueocean.js:84533)
    at ReactCompositeComponentWrapper.mountComponent (blueocean.js:84429)
    at Object.mountComponent (blueocean.js:91442)
    at ReactDOMComponent.mountChildren (blueocean.js:90320)
    at ReactDOMComponent._createInitialChildren (blueocean.js:85896)
    at ReactDOMComponent.mountComponent (blueocean.js:85721)
    at Object.mountComponent (blueocean.js:91442)
﻿

```

**Comment by michaelneale** [comment]
```
     displayName
                    ),
                    run.branch.url && _react2.default.createElement(
                        'a',
                        { className: 'inline-svg', title: 'Opens branch in a new window', target: '_blank', href: run.branch.url },
                        _react2.default.createElement(_reactMaterialIcons.Icon, { size: 14, icon: 'launch' })
                    )
                ) : _react2.default.createElement(
```

is where error is. Branch is null for some reason

File: blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx [hunk 1]
```diff
@@ -99,6 +99,11 @@ class RunDetailsHeader extends Component {
                 {isMultiBranch ? (
                     <span className={labelClassName}>
                         <Link to={ branchUrl }>{ displayName }</Link>
+                        { run.branch.url &&
```
Comment (i386) [inline blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx]: should be `run.branch && run.branch.url &&` to catch the undefined error?

Comment (NicuPascu) [inline blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx]: yup, thanks

**Comment by i386** [comment]
Ill give this a spin now

**Comment by i386** [comment]
Works for me :) feel free to merge when you get a successful test run 🐝

**Comment by NicuPascu** [comment]
Will do, thanks

**Comment by imeredith** [comment]
Branch is null for jobs like freestyle. need to do null check.

**Comment by michaelneale** [comment]
bringing this up to date with master and letting it spin up a run...

**Comment by michaelneale** [comment]
finally made it pass - took manh goes, which is a bit of a worry. I might recommend holding off on this until tomorrow, when we might have some light shed on the crapness that made it so hard to pass. Functionally great though, nice one.

**Comment by cliffmeyers** [comment]
Something with ATH has changed in the past few days (?) that is causing an incredibly high spurious failure rate. master is failing with some frequency.

**Comment by michaelneale** [comment]
ok updated it to master - if this passes. I think :bee: once that is green, nice work @NicuPascu !

**Comment by michaelneale** [comment]
:bee:

**Pull request merged**